### PR TITLE
add new receipt lines

### PR DESCRIPTION
**Pull request by mike-dydx**
## Links (dYdX Internal Use Only)
FIgma Design: https://www.figma.com/design/mKevZOfE9nj6MZpiolKYW1/dYdX-%E2%80%BA-Mobile?node-id=6314-5156&t=Akhipf3STQ4I0Wzs-4

<br/>

## Description / Intuition
- adds support for displaying three new receipt lines
- adds convenience "asNsNumber" for Int and Double

<br/>

## Before/After Screenshots or Videos

| Before | After |
|--------|-------|
|  |  |
|  |  |
|  |  |
|  |  |

<br/>

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring or Technical Debt
- [ ] Documentation update
- [ ] Other (please describe: )

### CODE REVIEW

File: dydx/dydxPresenters/dydxPresenters/_v4/Receipt/dydxTradeReceiptPresenter.swift [hunk 1]
```diff
@@ -80,10 +87,43 @@ final class dydxTradeReceiptPresenter: dydxReceiptPresenter {
...
+        let tickSize = market.configs?.displayTickSizeDecimals?.intValue.asNsNumber
+        liquidationPriceViewModel.title = title
+        liquidationPriceViewModel.value = createAmountChangeViewModel(title: title, tradeState: position?.liquidationPrice, tickSize: tickSize, unit: unit)
+    }
+
+    private func updatePositionMargin(position: SubaccountPosition?) {
+        let title = DataLocalizer.localize(path: "APP.TRADE.POSITION_MARGIN")
+        let unit = AmountTextModel.Unit.percentage
+        positionMarginViewModel.title = title
+        positionMarginViewModel.value = createAmountChangeViewModel(title: title, tradeState: position?.marginUsage, tickSize: 2, unit: unit)
```
Comment (ruixhuang) [inline dydx/dydxPresenters/dydxPresenters/_v4/Receipt/dydxTradeReceiptPresenter.swift:106]: Web shows Margin as the dollar value.  Let me add it to Abacus and we can change this later.

File: dydx/dydxPresenters/dydxPresenters/_v4/Trade/Margin/dydxTargetLeverageViewBuilder.swift [hunk 2]
```diff
@@ -95,8 +95,8 @@ private class dydxTargetLeverageViewPresenter: HostedViewPresenter<dydxTargetLev
                     let maxLeverage = 1.0 / imf
                     viewModel.leverageOptions = [1, 2, 5, 10]
                         .filter { $0 < maxLeverage }
```
Comment (ruixhuang) [inline dydx/dydxPresenters/dydxPresenters/_v4/Trade/Margin/dydxTargetLeverageViewBuilder.swift:97]: Web uses [1,2,3,5..10]

**Review by ruixhuang — APPROVED** [review]

File: dydx/dydxPresenters/dydxPresenters/_v4/Trade/Margin/dydxTargetLeverageViewBuilder.swift (hunk 2 above)
Comment (mike-dydx) [reply dydx/dydxPresenters/dydxPresenters/_v4/Trade/Margin/dydxTargetLeverageViewBuilder.swift:97]: will handle next

File: dydx/dydxPresenters/dydxPresenters/_v4/Receipt/dydxTradeReceiptPresenter.swift (hunk 1 above)
Comment (mike-dydx) [reply dydx/dydxPresenters/dydxPresenters/_v4/Receipt/dydxTradeReceiptPresenter.swift:106]: will handle next

**Pull request merged**